File: grimdb/sdk/swift/Sources/GrimlockerSDK/Models/FileEntry.swift
import Foundation

public struct FileEntry: Codable, Identifiable, Sendable {
    public let id: String
    public let fileName: String
    public let mimeType: String
    public let totalSize: Int64
    public let manifestBlockId: String
    public let folderId: String

    enum CodingKeys: String, CodingKey {
        case id
        case fileName = "file_name"
        case mimeType = "mime_type"
        case totalSize = "total_size"
        case manifestBlockId = "manifest_block_id"
        case folderId = "folder_id"
    }
}

File: grimdb/sdk/php/src/FileVault/FileEntry.php
<?php declare(strict_types=1);
namespace Grimlocker\FileVault;

class FileEntry
{
    public function __construct(
        public readonly string $id,
        public readonly string $fileName,
        public readonly string $mimeType,
        public readonly int    $totalSize,
        public readonly string $manifestBlockId,
        public readonly string $folderId,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            $data['id']                ?? '',
            $data['file_name']         ?? '',
            $data['mime_type']         ?? '',
            $data['total_size']        ?? 0,
            $data['manifest_block_id'] ?? '',
            $data['folder_id']         ?? '',
        );
    }
}
